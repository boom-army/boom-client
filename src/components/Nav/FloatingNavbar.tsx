import FeedIcon from "@mui/icons-material/Feed";
import HubIcon from "@mui/icons-material/Hub";
import NewspaperIcon from "@mui/icons-material/Newspaper";
import React from "react";
import ScienceIcon from "@mui/icons-material/Science";
import {
  Box,
  IconButton,
  Toolbar,
  useMediaQuery,
  useTheme,
} from "@mui/material";
import { RoutePath } from "../../constants";
import { useLocation, Link } from "react-router-dom";

interface NavIconButtonProps {
  icon: React.ReactNode;
  route: string;
}

const NavIconButton: React.FC<NavIconButtonProps> = ({ icon, route }) => {
  const theme = useTheme();
  const location = useLocation();
  const isActiveRoute = location.pathname === route;

  return (
    <Box
      sx={{ display: "flex", flexDirection: "column", alignItems: "center" }}
    >
      <Link to={route}>
        <IconButton
          color="inherit"
          sx={{
            "& svg": {
              color: isActiveRoute
                ? theme.accentColor
                : theme.palette.primary.main,
            },
            "&:hover": {
              "& svg": {
                color: theme.accentColor,
              },
            },
          }}
        >
          {icon}
        </IconButton>
      </Link>
    </Box>
  );
};

export const FloatingNavbar = () => {
  const theme = useTheme();
  const isDesktop = useMediaQuery(theme.breakpoints.up("md"));

  if (!isDesktop) {
    return null;
  }

  return (
    <Box
      sx={{
        position: "fixed",
        bottom: "5%",
        left: "50%",
        transform: "translateX(-50%)",
        zIndex: 1300,
        borderRadius: "50px",
        overflow: "hidden",
        width: { md: "40%", lg: "20%" },
        boxShadow: theme.shadows[3],
        backgroundColor: theme.tertiaryColor2,
        borderBottom: "1px solid rgba(0, 0, 0, 0.12)",
      }}
    >
      <Toolbar>
        <Box
          sx={{ display: "flex", flexGrow: 1, justifyContent: "space-evenly" }}
        >
          <NavIconButton icon={<HubIcon />} route={RoutePath.HOME} />
          <NavIconButton icon={<FeedIcon />} route={`/${RoutePath.FEED}`} />
          <NavIconButton
            icon={<NewspaperIcon />}
            route={`/${RoutePath.NEWS}`}
          />
          <NavIconButton icon={<ScienceIcon />} route={`/${RoutePath.LAB}`} />
        </Box>
      </Toolbar>
    </Box>
  );
};
