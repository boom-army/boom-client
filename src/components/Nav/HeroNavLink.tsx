import { alpha, Box, Stack, Typography, useTheme } from "@mui/material";
import { ReactElement } from "react";
import { NavLink } from "react-router-dom";
import { RoutePath } from "../../constants";

interface HeroNavLinkProps {
  icon: ReactElement;
  label: string;
  description: string;
  routePath: RoutePath;
}

export const HeroNavLink: React.FC<HeroNavLinkProps> = ({
  icon,
  label,
  description,
  routePath,
}) => {
  const theme = useTheme();

  const applyActiveStyles = ({ isActive }: { isActive: boolean }) => ({
    color: isActive
      ? theme.accentColor
      : (theme.palette.secondary.main),
    width: "100%",
  });

  const h5Styles = {
    my: 0.5,
  };
  const stackProps = {
    justifyContent: "flex-start",
    flexGrow: 1,
    spacing: 2,
    alignItems: "center",
    backgroundColor: theme.blue.darker,
    sx: {
      borderRadius: "3px",
      border: `1px solid ${alpha(theme.accentColor, 0.5)}`,
      width: "100%",
      p: 1,
      "&:hover": {
        backgroundColor: alpha(theme.accentColor, 0.2),
      },
    },
  };

  return (
    <NavLink end style={applyActiveStyles} to={routePath}>
      <Box {...stackProps}>
        <Stack direction="row" alignItems="top">
          <Box>{icon}</Box>
          <Box ml={1}>
            <Typography variant="h5" sx={h5Styles}>
              {label}
            </Typography>
            <Typography variant="body2" color="secondary">
              {description}
            </Typography>
          </Box>
        </Stack>
      </Box>
    </NavLink>
  );
};
