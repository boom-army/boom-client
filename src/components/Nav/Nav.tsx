import ArticleIcon from '@mui/icons-material/Article';
import FeedIcon from "@mui/icons-material/Feed";
import GitHubIcon from '@mui/icons-material/GitHub';
import HubIcon from "@mui/icons-material/Hub";
import NewspaperIcon from "@mui/icons-material/Newspaper";
import React from "react";
import ScienceIcon from "@mui/icons-material/Science";
import StarsIcon from '@mui/icons-material/Stars';
import SwapHorizontalCircleIcon from '@mui/icons-material/SwapHorizontalCircle';
import WorkspacePremiumIcon from '@mui/icons-material/WorkspacePremium';
import { ColorModeContext } from "../../contexts/theme";
import { HeroNavLink } from ".";
import { Logout } from "../Auth/Logout";
import { NavLink } from "react-router-dom";
import { RoutePath } from "../../constants";
import { Stack, Typography, useTheme } from "@mui/material";
import { User } from "../../generated/graphql";
import { styled } from "@mui/material/styles";

interface Props {
  newMentionsCount: number | undefined;
  user: User | null;
}

export const Nav: React.FC<Props> = ({ newMentionsCount, user }) => {
  const theme = useTheme();
  const { toggleColorMode } = React.useContext(ColorModeContext);

  const applyActiveStyles = ({ isActive }: { isActive: boolean }) => ({
    color: isActive
      ? theme.accentColor
      : (theme.palette.secondary as unknown as string),
    width: "100%",
  });

  const iconProps = {
    color: theme.accentColor,
  };

  const stackProps = {
    justifyContent: "flex-start",
    flexGrow: 1,
    spacing: 1,
    alignItems: "center",
    sx: {
      width: "100%",
    },
  };

  const StyledStack = styled(Stack)({
    paddingTop: "0.5rem",
    width: "95%",
    fontWeight: 500,
    alignItems: "flex-start",
    justifyContent: "space-between",
    height: "100vh",
    "@media screen and (max-width: 900px)": {
      minWidth: "150px",
    },
  });

  return (
    <StyledStack key="main-stack" direction="column" spacing={1}>
      <Stack key="top-nav-stack" direction="column" spacing={1}>
        <HeroNavLink
          icon={<HubIcon sx={iconProps} />}
          label="DAOs"
          description="DAOs are the heart of Boom. They are NFT gated communities made up of Solana's biggest fans."
          routePath={RoutePath.HOME}
        />
        <HeroNavLink
          icon={<FeedIcon sx={iconProps} />}
          label="Feed"
          description="The Feed is like Twitter where you can post Meeps by connecting your wallet. It's a great place to start."
          routePath={RoutePath.FEED}
        />
        <HeroNavLink
          icon={<NewspaperIcon sx={iconProps} />}
          label="News"
          description="News pulls in all the latest news from the Solana ecosystem. It's your place to keep up to date."
          routePath={RoutePath.NEWS}
        />
        <HeroNavLink
          icon={<ScienceIcon sx={iconProps} />}
          label="Lab"
          description="The Lab is where you can find all Boom's latest Solana experiments and tools."
          routePath={RoutePath.LAB}
        />
      </Stack>
      <Stack key="bottom-nav-stack" direction="column" spacing={1}>
        <NavLink style={applyActiveStyles} to={"https://www.tensor.trade/trade/boomheroes"} target="_blank">
          <Stack direction="row" {...stackProps}>
            <StarsIcon sx={iconProps} fontSize="small" />
            <Typography variant="body2" color="secondary">Buy a Boom DAO NFT</Typography>
          </Stack>
        </NavLink>
        <NavLink style={applyActiveStyles} to={"https://moonrank.app/collection/boom_heroes"} target="_blank">
          <Stack direction="row" {...stackProps}>
            <WorkspacePremiumIcon sx={iconProps} fontSize="small" />
            <Typography variant="body2" color="secondary">Boom Heroes Rarity</Typography>
          </Stack>
        </NavLink>
        <NavLink style={applyActiveStyles} to={"https://docs.boom.army/blog/"} target="_blank">
          <Stack direction="row" {...stackProps}>
            <ArticleIcon sx={iconProps} fontSize="small" />
            <Typography variant="body2" color="secondary">Blog</Typography>
          </Stack>
        </NavLink>
        <NavLink style={applyActiveStyles} to={"https://jup.ag/swap/USDC-BMA"} target="_blank">
          <Stack direction="row" {...stackProps}>
            <SwapHorizontalCircleIcon sx={iconProps} fontSize="small" />
            <Typography variant="body2" color="secondary">USDC-BMA</Typography>
          </Stack>
        </NavLink>
        <NavLink style={applyActiveStyles} to={"https://github.com/boom-army"} target="_blank">
          <Stack direction="row" {...stackProps}>
            <GitHubIcon sx={iconProps} fontSize="small" />
            <Typography variant="body2" color="secondary">Boom Github</Typography>
          </Stack>
        </NavLink>
      </Stack>
    </StyledStack>
  );
};
