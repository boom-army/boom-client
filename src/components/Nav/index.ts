export * from "./FloatingNavbar";
export * from "./HeroNavLink";
export * from "./LeftNavDrawer";
export * from "./MiniDrawer";
export * from "./MobileBottomNav";
export * from "./Nav";
export * from "./PopoutDrawerMenu";
