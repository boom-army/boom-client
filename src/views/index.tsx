export { BoomOnes } from "./BoomOnes";
export { ChannelFeed } from "./ChannelFeed";
export { Channels } from "./Channels";
export { PeopleView } from "./People";
export { Explore } from "./Explore";
export { Following } from "./Following";
export { Dashboard } from "./Dashboard";
export { News } from "./News";
export { HeroFeed } from "./HeroFeed";
// export { OGMint } from "./OGMint";
