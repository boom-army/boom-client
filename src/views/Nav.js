import React from "react";
import styled from "styled-components";
import { useQuery } from "@apollo/client";
import { NavLink } from "react-router-dom";
import { MorePopUp } from "../components/MorePopup";
import { PROFILE } from "../queries/profile";
import Person from "@mui/icons-material/Person";
import GroupIcon from "@mui/icons-material/Group";
import Language from "@mui/icons-material/Language";
import NotificationsIcon from "@mui/icons-material/Notifications";
import AccountCircleIcon from "@mui/icons-material/AccountCircle";

const Wrapper = styled.nav`
  font-weight: 500;
  min-height: 100vh;

  .badge {
    padding: 2px 7px;
    border-radius: 50%;
    background: ${(props) => props.theme.accentColor};
    color: white;
    top: -10px;
    position: relative;
    font-size: 0.7em;
  }

  svg {
    width: 28px;
    height: 28px;
    margin-right: 0.5rem;
    position: relative;
    color: ${(props) => props.theme.accentColor};
    top: 7px;
  }

  .logo {
    margin-bottom: 1.3rem;
  }

  ul {
    margin-top: 1.3rem;
    display: flex;
    flex-direction: column;
    height: 100%;
  }

  li {
    margin-bottom: 2rem;
  }

  .selected,
  .selected svg {
    color: ${(props) => props.theme.accentColor};
    fill: ${(props) => props.theme.accentColor};
  }

  @media screen and (max-width: 900px) {
    span {
      display: none;
    }

    svg {
      margin-right: 0;
    }

    li {
      margin: none;
    }

    button {
      display: none;
    }
  }

  @media screen and (max-width: 530px) {
    bottom: 0;
    width: 100vw;
    border-right: none;
    height: 4rem;
    border-top: 1px solid ${(props) => props.theme.tertiaryColor};
    z-index: 2;
    background: ${(props) => props.theme.background};

    ul {
      flex-direction: row;
      justify-content: space-between;
    }

    li {
    }

    svg {
      width: 22px;
      height: 22px;
    }
  }

  @media screen and (max-width: 500px) {
  }
`;

export const Nav = () => {
  const user = JSON.parse(localStorage.getItem("user"));
  const { data } = useQuery(PROFILE, {
    variables: { id: user?.id },
  });

  return (
    <Wrapper>
      <ul>
        <li>
          <NavLink exact activeClassName="selected" to="/">
            <Language /> <span>Global</span>
          </NavLink>
        </li>
        <li>
          <NavLink activeClassName="selected" to="/following">
            <Person /> <span>Following</span>
          </NavLink>
        </li>
        <li>
          <NavLink activeClassName="selected" to="/connect">
            <GroupIcon /> <span>Connect</span>
          </NavLink>
        </li>
        <li>
          <NavLink activeClassName="selected " to="/notifications">
            <NotificationsIcon /> <span>Notifications</span>
            {data?.profile?.newMentionsCount >= 1 && (
              <span className="badge">{data?.profile?.newMentionsCount}</span>
            )}
          </NavLink>
        </li>
        <li>
          <NavLink activeClassName="selected" to="/profile">
            <AccountCircleIcon /> <span>Profile</span>
          </NavLink>
        </li>
        <li>
          <MorePopUp />
        </li>
        {/* <li> */}
        {/* 	<Logout /> */}
        {/* </li> */}
        {/* <li style={{ display: "" }}> */}
        {/* 	<ToggleTheme /> */}
        {/* </li> */}
      </ul>
    </Wrapper>
  );
};
