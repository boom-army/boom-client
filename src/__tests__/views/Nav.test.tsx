import React from "react";
import { Nav } from "../../views/Nav";
import { act, fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

let user: any = {
  id: "ckzny1iv10031lmn568gj3b3n",
  publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
  avatar: "",
  handle: "shy-cloud-4965",
  consumerName: "shy-cloud-4965",
  __typename: "User",
};

describe("<Nav/> component :", () => {
  test("display <Nav/> component ", async () => {
    let rendered = render(<Nav newMentionsCount={1} user={user} />, {
      wrapper: MemoryRouter,
    });
    expect(rendered).toMatchSnapshot();
  });
});

test("Assertion testing of <Nav/> component", () => {
  render(<Nav newMentionsCount={1} user={user} />, {
    wrapper: MemoryRouter,
  });

  expect(screen.getByRole("link", { name: "Community" })).toHaveAttribute(
    "href",
    "/"
  );
  expect(screen.getByRole("link", { name: "Mint NFT" })).toHaveAttribute(
    "href",
    "/mint-nft"
  );
  expect(screen.getByRole("link", { name: "Creators" })).toHaveAttribute(
    "href",
    "/connect"
  );
  expect(screen.getByRole("link", { name: "1 Notifications" })).toHaveAttribute(
    "href",
    "/notifications"
  );
  expect(screen.getByRole("link", { name: "Profile" })).toHaveAttribute(
    "href",
    "/shy-cloud-4965"
  );
});
