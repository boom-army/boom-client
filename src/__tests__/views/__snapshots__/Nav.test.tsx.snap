// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav/> component : display <Nav/> component  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1h9n16o-MuiStack-root"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
          style="color: rgb(202, 32, 85);"
        >
          <div
            class="css-yhjy8k-MuiStack-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LanguageIcon"
              focusable="false"
              style="color: rgb(202, 32, 85);"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Community
            </p>
          </div>
        </a>
        <a
          class=""
          href="/mint-nft"
          style="color: rgb(255, 255, 255);"
        >
          <div
            class="css-yhjy8k-MuiStack-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3l9bez-MuiSvgIcon-root"
              data-testid="StyleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m2.53 19.65 1.34.56v-9.03l-2.43 5.86c-.41 1.02.08 2.19 1.09 2.61zm19.5-3.7L17.07 3.98c-.31-.75-1.04-1.21-1.81-1.23-.26 0-.53.04-.79.15L7.1 5.95c-.75.31-1.21 1.03-1.23 1.8-.01.27.04.54.15.8l4.96 11.97c.31.76 1.05 1.22 1.83 1.23.26 0 .52-.05.77-.15l7.36-3.05c1.02-.42 1.51-1.59 1.09-2.6zM7.88 8.75c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm-2 11c0 1.1.9 2 2 2h1.45l-3.45-8.34v6.34z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Mint NFT
            </p>
          </div>
        </a>
        <a
          class=""
          href="/connect"
          style="color: rgb(255, 255, 255);"
        >
          <div
            class="css-yhjy8k-MuiStack-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3l9bez-MuiSvgIcon-root"
              data-testid="GroupIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Creators
            </p>
          </div>
        </a>
        <a
          class=""
          href="/notifications"
          style="color: rgb(255, 255, 255);"
        >
          <div
            class="css-yhjy8k-MuiStack-root"
          >
            <span
              class="MuiBadge-root css-15rhyob-MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3l9bez-MuiSvgIcon-root"
                data-testid="NotificationsIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
                />
              </svg>
              <span
                class="MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight css-fvc8ir-MuiBadge-badge"
              >
                1
              </span>
            </span>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Notifications
            </p>
          </div>
        </a>
        <a
          class=""
          href="/shy-cloud-4965"
          style="color: rgb(255, 255, 255);"
        >
          <div
            class="css-yhjy8k-MuiStack-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3l9bez-MuiSvgIcon-root"
              data-testid="AccountCircleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Profile
            </p>
          </div>
        </a>
        <div
          class="css-1vkxm47"
        >
          <div
            aria-describedby="popup-1"
          >
            <div
              class="css-fcjwbd-MuiStack-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3l9bez-MuiSvgIcon-root"
                data-testid="MoreHorizIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body1 css-6p3zyw-MuiTypography-root"
              >
                More
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1h9n16o-MuiStack-root"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
        style="color: rgb(202, 32, 85);"
      >
        <div
          class="css-yhjy8k-MuiStack-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LanguageIcon"
            focusable="false"
            style="color: rgb(202, 32, 85);"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Community
          </p>
        </div>
      </a>
      <a
        class=""
        href="/mint-nft"
        style="color: rgb(255, 255, 255);"
      >
        <div
          class="css-yhjy8k-MuiStack-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3l9bez-MuiSvgIcon-root"
            data-testid="StyleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m2.53 19.65 1.34.56v-9.03l-2.43 5.86c-.41 1.02.08 2.19 1.09 2.61zm19.5-3.7L17.07 3.98c-.31-.75-1.04-1.21-1.81-1.23-.26 0-.53.04-.79.15L7.1 5.95c-.75.31-1.21 1.03-1.23 1.8-.01.27.04.54.15.8l4.96 11.97c.31.76 1.05 1.22 1.83 1.23.26 0 .52-.05.77-.15l7.36-3.05c1.02-.42 1.51-1.59 1.09-2.6zM7.88 8.75c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm-2 11c0 1.1.9 2 2 2h1.45l-3.45-8.34v6.34z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Mint NFT
          </p>
        </div>
      </a>
      <a
        class=""
        href="/connect"
        style="color: rgb(255, 255, 255);"
      >
        <div
          class="css-yhjy8k-MuiStack-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3l9bez-MuiSvgIcon-root"
            data-testid="GroupIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Creators
          </p>
        </div>
      </a>
      <a
        class=""
        href="/notifications"
        style="color: rgb(255, 255, 255);"
      >
        <div
          class="css-yhjy8k-MuiStack-root"
        >
          <span
            class="MuiBadge-root css-15rhyob-MuiBadge-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3l9bez-MuiSvgIcon-root"
              data-testid="NotificationsIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
              />
            </svg>
            <span
              class="MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight css-fvc8ir-MuiBadge-badge"
            >
              1
            </span>
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Notifications
          </p>
        </div>
      </a>
      <a
        class=""
        href="/shy-cloud-4965"
        style="color: rgb(255, 255, 255);"
      >
        <div
          class="css-yhjy8k-MuiStack-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3l9bez-MuiSvgIcon-root"
            data-testid="AccountCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Profile
          </p>
        </div>
      </a>
      <div
        class="css-1vkxm47"
      >
        <div
          aria-describedby="popup-1"
        >
          <div
            class="css-fcjwbd-MuiStack-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3l9bez-MuiSvgIcon-root"
              data-testid="MoreHorizIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 css-6p3zyw-MuiTypography-root"
            >
              More
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
